suite-native/module-add-coin/src/hooks/useAddCoinAccount.ts

```diff
@@ -0,0 +1,250 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { TrezorDevice } from '@suite-common/suite-types';
+import { AccountType, NetworkSymbol, Network } from '@suite-common/wallet-config';
+import {
+    AccountsRootState,
+    DeviceRootState,
+    selectDevice,
+    selectDeviceAccounts,
+} from '@suite-common/wallet-core';
+import { Account } from '@suite-common/wallet-types';
+import { useAlert } from '@suite-native/alerts';
+import {
+    addNetworkAccountThunk,
+    selectAreTestnetsEnabled,
+    selectDiscoverySupportedNetworks,
+    getCardanoSupportedAccountTypesThunk,
+} from '@suite-native/discovery';
+import { useTranslate } from '@suite-native/intl';
+import { useOpenLink } from '@suite-native/link';
+import {
+    RootStackParamList,
+    RootStackRoutes,
+    StackNavigationProps,
+} from '@suite-native/navigation';
+
+const availableNetworkAccountTypes: Map<NetworkSymbol | undefined, AccountType[]> = new Map([
+    ['btc', ['normal', 'taproot', 'segwit', 'legacy']],
+    ['ltc', ['normal', 'segwit', 'legacy']],
+    ['btg', ['normal', 'segwit', 'legacy']],
+    ['dgb', ['normal', 'segwit', 'legacy']],
+    ['vtc', ['normal', 'segwit', 'legacy']],
+]);
+
+export const useAddCoinAccount = () => {
+    const dispatch = useDispatch();
+    const { translate } = useTranslate();
+    const openLink = useOpenLink();
+    const areTestnetsEnabled = useSelector(selectAreTestnetsEnabled);
+    const supportedNetworks = useSelector((state: DeviceRootState) =>
+        selectDiscoverySupportedNetworks(state, areTestnetsEnabled),
+    );
+    const accounts = useSelector((state: AccountsRootState & DeviceRootState) =>
+        selectDeviceAccounts(state),
+    );
+    const device = useSelector(selectDevice);
+    const { showAlert, hideAlert } = useAlert();
+    const navigation =
+        useNavigation<StackNavigationProps<RootStackParamList, RootStackRoutes.AccountDetail>>();
+    const [availableCardanoTypes, setAvailableCardanoTypes] = useState<AccountType[]>([]);
+    const [supportedNetworkSymbols, setSupportedNetworkSymbols] = useState<NetworkSymbol[]>([]);
```

Comment:
Why do we need this? We have supportedNetworkSymbols available through selectors and it shouldn't be duplicated here